apps/stripe/src/app/api/saleor/transaction-initialize-session/use-case.test.ts

```diff
@@ -0,0 +1,101 @@
+import { err, ok } from "neverthrow";
+import { describe, expect, it, vi } from "vitest";
+
+import { mockedStripeConfig } from "@/__tests__/mocks/stripe-config";
+import { MockedStripePaymentIntentsApi } from "@/__tests__/mocks/stripe-payment-intents-api";
+import { mockedTransactionInitializeSessionEvent } from "@/__tests__/mocks/transaction-initalize-session-event";
+import { InitializeStripeTransactionUseCase } from "@/app/api/saleor/transaction-initialize-session/use-case";
+import { BaseError } from "@/lib/errors";
+import { AppConfigPersistor } from "@/modules/app-config/app-config-persistor";
+import { SaleorApiUrl } from "@/modules/saleor/saleor-api-url";
+import { IStripePaymentIntentsApiFactory } from "@/modules/stripe/types";
+
+const saleorApiUrlMock = SaleorApiUrl.create({
+  url: "https://foo.bar.saleor.cloud/graphql/",
+})._unsafeUnwrap();
+const channelIdMock = "test-id";
+const mockAppId = "test-id";
+
+describe("InitializeStripeTransactionUseCase", () => {
+  const testConfig: AppConfigPersistor = {
+    getStripeConfig: vi.fn(),
+    saveStripeConfig: vi.fn(),
+  };
+  const testStripeApiFactory: IStripePaymentIntentsApiFactory = {
+    create: vi.fn(),
+  };
```

Comment:
you create it only once, and you even forgot to initialize it in one test. maybe just create it's plain implementation here? I don't see benefit of mocking factory - this is comfort of dependency injection that it's rarely needed